### ISSUE TITLE
Not being able to save a broken function is super frustrating, especially when error detection is handled by the bug catcher. 

### ISSUE THREAD

**Issue by chuggmuffinReturns** [FlutterFlow/flutterflow-issues#5054]
### Can we access your project?

- [X] I give permission for members of the FlutterFlow team to access and test my project for the sole purpose of investigating this issue.

### Current Behavior

https://www.loom.com/share/f77a77bac74e4673bf6e52cfd9be6a25

When you have a broken function it prevents you from saving which makes checking the function against other code impossible. 

So you have to comment things out and then create a dummy return statement in order to save and that sucks because you cant highlight and comment out a whole bunch of text like in VS code so THAT involves commenting out each line which clearly is never going to happen. 

### Expected Behavior

Ill call this "preferred" behaviour...

Allow saving of broken functions and let the bug detector do its job

### Steps to Reproduce

create function
make it stupid
hit save
see prevention of it being saved

### Reproducible from Blank

- [X] The steps to reproduce above start from a blank project.

### Bug Report Code (Required)

https://www.loom.com/share/f77a77bac74e4673bf6e52cfd9be6a25

### Visual documentation

https://www.loom.com/share/f77a77bac74e4673bf6e52cfd9be6a25

### Environment

```markdown
mac desktop latest
```

**Comment by Alezanello** [comment]
Hello!

There are many cases where the system allows you to save a function even if the code inside contains errors. The only time it prevents you from saving is when the code is completely broken and cannot be parsed, such as when a closing bracket `}` is missing.

In the example you shared, the message "Is empty or cannot be parsed" indicates that as long as the code can be parsed—even if it contains logical errors—it can still be saved. The missing closing bracket `}` in your example would prevent it from being parsed.

To summarize, there are two levels of checks in place:

1. **Parsing Check (Save Validation):** The function must not be empty and must be parseable. This is the only requirement for saving the function.
2. **Syntax Check (Execution Validation):** This ensures the code inside the function is correctly written to be executed or tested. This validation occurs when you run or compile the project.

**Comment by chuggmuffinReturns** [comment]
hi @Alezanello bit of a longer response this time...

There are few major problems with the approach as it currently stands...

1) when you're creating a new function you're trying to solve the business problem and referencing other functions in order to build the new function. 

Forcing code to always be in a parsable state really interrupts the process of code assembly and interrupts the thought process and causes me, at least, to often lose train of thought on the original problem. 

2) not being able to save the function means you can't download it to let VS Code debug it, further requiring brain power to be devoted to finding a ; or whatever and causing loss of train of thought

3) not being able to bulk comment out code makes it impossible to comment out, add in a dummy return statement, to then save it to then download and debug

...and wahts most frustrating...

4) It doesn't even matter! You catch logic errors later in the process and prevent compilation so why interrupt iterative workflow for something that seems to be completely arbitrary. 

So why not let me save anyway, maybe throw an asterisk somewhere or a sentence like you do with "restore unsaved changes". 

I am probably going to introduce another parsing problem later, or fixing a logic problem is going to eliminate the source of the parsing problem anyway.

**Comment by Alezanello** [comment]
I'll talk to the team about this answer and come back to you with more updates!<br>Have a nice weekend!

**Comment by Alezanello** [comment]
Quick update:<br><br>The team is already working on a way to save the functions even if they have errors!

**Comment by chuggmuffinReturns** [comment]
Thanks a lot!

I have already seen some small changes regarding that recently so that is encouraging. 

Will make things a lot easier!

**Comment by chuggmuffinReturns** [comment]
I appreciate the update as well :)
